tss-esapi/examples/aes_encrypt_decrypt_rust_crypto.rs

```diff
@@ -0,0 +1,312 @@
+
+    if k_byte as usize > AES_BLOCK_SIZE {
+        eprintln!("{}", k_byte);
+        panic!("Invalid pad byte, exceeds AES_BLOCK_SIZE");
+    }
+
+    for i in 0..k_byte {
+        if decrypted_data[last_byte - i as usize] != k_byte {
+            panic!("Invalid pad byte, is not equal to k_byte");
+        }
+    }
+
+    let truncate_to = decrypted_data.len().checked_sub(k_byte as usize).unwrap();
+    let mut decrypted_data = decrypted_data.to_vec();
+    decrypted_data.truncate(truncate_to);
+
+    // END WARNING
+
+    println!("data_to_encrypt = {:?}", data_to_encrypt);
+    println!("decrypted_data = {:?}", decrypted_data);
+    // They are the same!
+    assert_eq!(data_to_encrypt, decrypted_data);
+}
+
+fn create_primary(context: &mut Context) -> CreatePrimaryKeyResult {
+    let object_attributes = ObjectAttributesBuilder::new()
+        .with_fixed_tpm(true)
+        .with_fixed_parent(true)
+        .with_st_clear(false)
+        .with_sensitive_data_origin(true)
+        .with_user_with_auth(true)
+        .with_decrypt(true)
+        .with_restricted(true)
+        .build()
+        .expect("Failed to build object attributes");
+
+    let primary_pub = PublicBuilder::new()
+        .with_public_algorithm(PublicAlgorithm::SymCipher)
+        .with_name_hashing_algorithm(HashingAlgorithm::Sha256)
+        .with_object_attributes(object_attributes)
+        .with_symmetric_cipher_parameters(SymmetricCipherParameters::new(
+            SymmetricDefinitionObject::AES_128_CFB,
+        ))
+        .with_symmetric_cipher_unique_identifier(Digest::default())
+        .build()
+        .unwrap();
+
+    context
+        .execute_with_nullauth_session(|ctx| {
+            ctx.create_primary(Hierarchy::Owner, primary_pub, None, None, None, None)
+        })
+        .unwrap()
+}
+
+
+// In the future I would probably do this where the object is actually a
+// stored context that is loaded/unloaded as required. We would also check
+// the handle is suitable for this purpose etc.
+struct TpmAes128CbcEnc<'a> {
+    cipher: TpmEnc<'a>,
+}
+
+struct TpmEnc<'a> {
+    ctx: &'a mut Context,
+    handle: KeyHandle,
+    iv: InitialValue,
+}
+
+impl<'a> cipher::BlockSizeUser for TpmAes128CbcEnc<'a> {
+    type BlockSize = cipher::consts::U16;
+}
+
+impl<'a> cipher::BlockEncryptMut for TpmAes128CbcEnc<'a> {
+    fn encrypt_with_backend_mut(
+        &mut self, 
+        f: impl cipher::BlockClosure<BlockSize = Self::BlockSize>
+    ) {
+        // No errors? I'll raise an issue.
+        let Self { cipher } = self;
+
+        f.call(cipher)
+    }
+}
+
+impl<'a> cipher::BlockSizeUser for TpmEnc<'a> {
+    type BlockSize = cipher::consts::U16;
+}
+
+impl<'a> cipher::ParBlocksSizeUser for TpmEnc<'a> {
+    // 1024 / 16 bytes
+    type ParBlocksSize = cipher::consts::U16;
+}
+
+impl<'a> cipher::BlockBackend for TpmEnc<'a> {
+    fn proc_block(&mut self, mut block: cipher::inout::InOut<'_, '_, cipher::Block<Self>>) {
```

Comment:
I guess `InOut` needs to be added [here](https://github.com/parallaxsecond/rust-tss-esapi/blob/main/.codespellrc#L3) :roll_eyes: